pkg/cmd/cli/cmd/logs.go

```diff
@@ -189,5 +218,43 @@ func (o OpenShiftLogsOptions) RunLog() error {
 		// Use our own options object.
 		o.KubeLogOptions.Options = o.Options
 	}
-	return o.KubeLogOptions.RunLogs()
+	if !o.JenkinsLogs {
+		return o.KubeLogOptions.RunLogs()
+	}
+
+	bc, ok := o.KubeLogOptions.Object.(*buildapi.BuildConfig)
+	var urlString, buildName string
+	if !ok {
+		build, ok := o.KubeLogOptions.Object.(*buildapi.Build)
+		if ok {
+			buildName = build.ObjectMeta.Name
+			urlString, _ = build.Annotations[buildapi.BuildJenkinsBlueOceanLogURLAnnotation]
+			if len(urlString) == 0 {
+				return errors.New(fmt.Sprintf("The build %s did not have the correct log URL", buildName))
+			}
+		}
+	} else {
+		buildName = buildutil.BuildNameForConfigVersion(bc.ObjectMeta.Name, int(bc.Status.LastVersion))
+		build, _ := o.Client.Builds(o.Namespace).Get(buildName, metav1.GetOptions{})
+		if build == nil {
+			return errors.New(fmt.Sprintf("The build %s for build config %s was not found", buildName, bc.ObjectMeta.Name))
+		}
+		urlString, _ = build.Annotations[buildapi.BuildJenkinsBlueOceanLogURLAnnotation]
+		if len(urlString) == 0 {
+			return errors.New(fmt.Sprintf("The build %s did not have the correct log URL", buildName))
+		}
```

Comment:
I added this in an attempt to address your previous request to be more precise about the errors.  For example, a build existed but did not have the annotation vs. say not being able to find the build object for the latest version noted in the build config status.

Pending any clarification on that interpretation, I'll get started with the rework you noted in your next comment, and see how that influences the specific errors returned like the one here.